gorilla/example_test.go

```diff
@@ -0,0 +1,58 @@
+package gorilla_test
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/opentracing-contrib/go-gorilla/gorilla"
+	"github.com/uber/jaeger-client-go/config"
+)
+
+func ExampleMiddleware() {
+	// http listen port
+	httpAddress := ":8800"
+
+	//jaeger agent port
+	jaegerHostPort := ":6831"
+	myHandler := func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		idstr := vars["productId"]
+		//do something
+		data := "Hello, we get " + idstr
+		fmt.Fprintf(w, data)
+	}
+	cfg := config.Configuration{
+		Sampler: &config.SamplerConfig{
+			Type:  "const",
+			Param: 1,
+		},
+		Reporter: &config.ReporterConfig{
+			LogSpans:            false,
+			BufferFlushInterval: 1 * time.Second,
+			LocalAgentHostPort:  jaegerHostPort,
+		},
+	}
+	tracer, _, err := cfg.New(
+		"ExampleMiddleware", //operationName
+	)
+
+	if err != nil {
+		log.Fatal("cannot initialize Jaeger Tracer", err)
+	}
+
+	r := mux.NewRouter()
+
+	pattern := "/v1/products/{productId}"
+	middleware := gorilla.Middleware(
```

Comment:
what's the reason to use `gorilla.Middleware` instead of `nethttp.Middleware` from [go-stdlib][1]?

I thought the whole idea was to use some internal knowledge of Gorilla (perhaps encoded in `http.Request` or the context) to extract the operation name equal to the route. This example looks like everything is done manually, so I don't know what it adds over go-stdlib module.

[1]: https://github.com/opentracing-contrib/go-stdlib/blob/master/nethttp/server.go#L77

Reply:
Yes, in fact, there is slight change from go-stdlib. we can give OperationNameFunc a default impl in function Middleware, with knowledge of Gorilla/context.  Will update today.  Thanks！